Bottom: 76641057dbd6999a0e712337b3fa1c9f4162ec52
Top:    aac98239f7019abe974867007ae1e059d5c62b01
Author: Peter Clifton <peter@clifton-electronics.co.uk>
Date:   2015-01-01 17:11:11 +0000

Fix to ParseGroupString()


---

diff --git a/src/misc.c b/src/misc.c
index 10796fc..63d7ca4 100644
--- a/src/misc.c
+++ b/src/misc.c
@@ -1187,7 +1187,14 @@ ParseGroupString (char *group_string, LayerGroupType *LayerGroup, int *LayerN)
           /* ignore white spaces and check for separator */
           while (*s && isspace ((int) *s))
             s++;
-          if (!*s || *s == ':')
+
+          /* Backup so the outer loop terminates on the NULL termination */
+          if (*s == '\0') {
+            s--;
+            break;
+          }
+
+          if (*s == ':')
             break;
         }
       LayerGroup->Number[group] = member;
